lib/theme_settings_object_validator.rb

```diff
@@ -163,9 +163,13 @@ def has_valid_property_value?(property_attributes, property_name)
         return false
       end
 
-      if validations&.dig(:url) && !value.match?(URI.regexp)
-        add_error(property_name, :string_value_not_valid_url)
-        return false
+      if validations&.dig(:url)
+        begin
+          URI.parse(value)
+        rescue URI::Error
```

Comment:
You can skip this by doing `UrlHelper.relaxed_parsed`:

https://github.com/discourse/discourse/blob/main/lib/url_helper.rb#L10-L25

Then you can just add error + return false if the return value is `nil`

Reply:
O nice 👍 Have updated the PR.